docs/source/notes/distributed_training.rst

```diff
@@ -88,6 +88,26 @@ accumulation, and vice versa. Our trainer strives to respect this equivalency
 and ensure identical behavior regardless of the combinations of space and time
 parallelization used.
 
+Distributed Sampling
+--------------------
+
+When providing :class:`torch.utils.data.Dataset` which is not :class:`torch.utils.data.IterableDataset`
+with :class:`torch.utils.data.DataLoader` to Composer, a :class:`torch.utils.data.distributed.DistributedSampler`
+is necessary to ensure different devices receive different batches. Composer will
+raise an error if a DistributedSampler is not provided. :mod:`composer.utils.dist`
+provides a helper function to create a DistributedSampler with the correct
+parameters in :meth:`composer.utils.dist.get_sampler`.
+
+.. code:: python
+
+    from composer.utils import dist
+
+    sampler = dist.get_sampler(dataset, shuffle=True)
+
+    dataloader = DataLoader(dataset, batch_size=32, sampler=sampler)
+
+:class:`composer.datasets.StreamingDataset` is an IterableDataset, so a
```

Comment:
Based on my understanding, this is a little bit misleading (as is the above mention of not being an `IterableDataset`), because `IterableDataset`s are expected to handle the "iterate differently on each GPU" themselves. They just don't work with samplers. See https://github.com/pytorch/pytorch/issues/28743 and https://pytorch.org/docs/stable/data.html#torch.utils.data.IterableDataset.

Reply:
hm... can you elaborate what you'd want it to say? a bit confused

Reply:
maybe "StreamingDataset is an IterableDataset, and therefore needs to handle multi-worker training itself (see https://pytorch.org/docs/stable/data.html#torch.utils.data.IterableDataset for more details)"

Reply:
":class:`composer.datasets.StreamingDataset` is an IterableDataset so a
DistributedSampler is not required as IterableDatasets need to handle multi-worker
training internally."

thoughts

Reply:
*DistributedSampler is not supported

Because the sampler doesn't even work with `IterableDataset` (https://github.com/pytorch/pytorch/issues/28743).

I would also leave the link to `IterableDataset` docs